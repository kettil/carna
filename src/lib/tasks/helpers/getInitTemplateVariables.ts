import { TemplateVariable } from '../../actions/node/template';
import { InitSettings } from './getInitSettings';

const getInitTemplateVariables = (settings: Pick<InitSettings, 'github'>): Record<string, TemplateVariable> => {
  /* eslint-disable @typescript-eslint/naming-convention */
  const variables: Record<string, TemplateVariable> = {
    GITHUB_USERNAME: settings.github.name,
  };

  return variables;
};

export default getInitTemplateVariables;
