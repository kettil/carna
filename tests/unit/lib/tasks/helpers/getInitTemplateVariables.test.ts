import getInitTemplateVariables from '../../../../../src/lib/tasks/helpers/getInitTemplateVariables';

describe('getTemplateVariables()', () => {
  const params: Array<[boolean, boolean, string | undefined]> = [
    [false, false, undefined],
    [false, false, 'username'],
    [false, true, undefined],
    [false, true, 'username'],
    [true, false, undefined],
    [true, false, 'username'],
    [true, true, undefined],
    [true, true, 'username'],
  ];

  test.each(params)('it should work (cli: %p, package: %p, github: %p)', (cli, isPackage, github) => {
    const settings = getInitTemplateVariables({ cli, package: isPackage }, { github: { name: github } });

    expect(settings).toMatchSnapshot();
  });
});
