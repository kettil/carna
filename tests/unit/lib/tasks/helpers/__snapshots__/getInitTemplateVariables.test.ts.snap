// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`getTemplateVariables() it should work (cli: false, package: false, github: "username") 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": "username",
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: false, package: false, github: undefined) 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": undefined,
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: false, package: true, github: "username") 1`] = `
Object {
  "BABEL_MODULE_TARGET": Object {
    "node": "14",
  },
  "BABEL_MODULE_TYPE": false,
  "GITHUB_USERNAME": "username",
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: false, package: true, github: undefined) 1`] = `
Object {
  "BABEL_MODULE_TARGET": Object {
    "node": "14",
  },
  "BABEL_MODULE_TYPE": false,
  "GITHUB_USERNAME": undefined,
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: true, package: false, github: "username") 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": "username",
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: true, package: false, github: undefined) 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": undefined,
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: true, package: true, github: "username") 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": "username",
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;

exports[`getTemplateVariables() it should work (cli: true, package: true, github: undefined) 1`] = `
Object {
  "BABEL_MODULE_TARGET": "defaults",
  "BABEL_MODULE_TYPE": "commonjs",
  "GITHUB_USERNAME": undefined,
  "PACKAGE_FILENAME": "packageName",
  "PACKAGE_LIBRARY": "packageName",
}
`;
